chaoscenter/graphql/server/graph/probe.resolvers.go

```diff
@@ -0,0 +1,262 @@
+
+import (
+	"context"
+	"errors"
+
+	"github.com/litmuschaos/litmus/chaoscenter/graphql/server/graph/model"
+	"github.com/litmuschaos/litmus/chaoscenter/graphql/server/pkg/authorization"
+	"github.com/litmuschaos/litmus/chaoscenter/graphql/server/pkg/chaos_experiment/handler"
+	"github.com/litmuschaos/litmus/chaoscenter/graphql/server/pkg/probe"
+	"github.com/sirupsen/logrus"
+)
+
+func (r *mutationResolver) AddProbe(ctx context.Context, request model.ProbeRequest, projectID string) (*model.Probe, error) {
+	logFields := logrus.Fields{
+		"projectId": projectID,
+	}
+
+	logrus.WithFields(logFields).Info("request received to create a probe")
+
+	err := authorization.ValidateRole(ctx, projectID,
+		authorization.MutationRbacRules[authorization.AddProbe],
+		model.InvitationAccepted.String())
+	if err != nil {
+		return nil, err
+	}
+
+	// Validate if probe type and probe properties match for the selected probe
+	// HTTP Probe type and Property validation
+	if request.Type == model.ProbeTypeHTTPProbe && request.HTTPProperties == nil {
+		err := "probe type and properties don't match, selected http Probe but http properties are empty"
+		logrus.WithFields(logFields).Error(err)
+		return nil, errors.New(err)
+	}
+	// PROM Probe type and Property validation
+	if request.Type == model.ProbeTypePromProbe && request.PromProperties == nil {
+		err := "probe type and properties don't match, selected prom Probe but prom properties are empty"
+		logrus.WithFields(logFields).Error(err)
+		return nil, errors.New(err)
+	}
+	// CMD Probe type and Property validation
+	if request.Type == model.ProbeTypeCmdProbe && request.CmdProperties == nil {
+		err := "probe type and properties don't match, selected cmd Probe but cmd properties are empty"
+		logrus.WithFields(logFields).Error(err)
+		return nil, errors.New(err)
+	}
+	// K8S Probe type and Property validation
+	if request.Type == model.ProbeTypeK8sProbe && request.K8sProperties == nil {
+		err := "probe type and properties don't match, selected k8s Probe but k8s properties are empty"
+		logrus.WithFields(logFields).Error(err)
+		return nil, errors.New(err)
+	}
+	p := probe.NewProbeRepository(projectID)
+	response, err := p.AddProbe(ctx, request)
+	if err != nil {
+		logrus.WithFields(logFields).Error(err)
+		return nil, err
+	}
+
+	return response, err
+}
+
+func (r *mutationResolver) UpdateProbe(ctx context.Context, request model.ProbeRequest, projectID string) (string, error) {
+	logFields := logrus.Fields{
+		"projectId": projectID,
+		"probeName": request.Name,
+	}
+
+	logrus.WithFields(logFields).WithField("probeID", request.Name).Info("request received to update probe")
+
+	err := authorization.ValidateRole(ctx, projectID,
+		authorization.MutationRbacRules[authorization.UpdateProbe],
+		model.InvitationAccepted.String())
+	if err != nil {
+		return "", err
+	}
+
+	p := probe.NewProbeRepository(projectID)
+	response, err := p.UpdateProbe(ctx, request)
```

Comment:
Probably needs same probe validity checks as done in addProbe

Reply:
I think we are still missing the part where we validate the probe type and the probe properties set match, for eg. if a user updates the probe but removes the probe properties that will be a problem and should be caught.